components/bio-formats-plugins/src/loci/plugins/util/ROIHandler.java

```diff
@@ -311,19 +337,48 @@ else if(ijRoi.getTypeAsString().matches("Rectangle")){//Check if its a Rectangle
 
     // -- Helper methods --
 
+    private static void storeText(TextRoi roi, MetadataStore store, int roiNum, int shape) {
+
+        store.setLabelX(roi.getPolygon().getBounds().getX(), roiNum, shape);
+        store.setLabelY(roi.getPolygon().getBounds().getY(), roiNum, shape);
+
+        store.setLabelText(roi.getText().trim(), roiNum, shape);
+        if (roi.getStrokeWidth() > 0) {
+            store.setLabelStrokeWidth( new Length((roi.getStrokeWidth()), UNITS.PIXEL), roiNum, shape);
+        }
+        if (roi.getStrokeColor() != null) {
+            store.setLabelStrokeColor(new ome.xml.model.primitives.Color(roi.getStrokeColor().getColorSpace().getType()) , roiNum, shape);
```

Comment:
This and the other places where store.set*Color are called won't preserve the ROI's color - ```getColorSpace().getType()``` on a ```java.awt.Color``` returns the type of the ColorSpace (e.g. RGB), not the color itself.

The easiest thing to do is to define a method that turns a java.awt.Color into an ome.xml.model.primitives.Color by passing the getRed(), getGreen(), getBlue(), and getAlpha() values on java.awt.Color to the OME Color's constructor.  That method can then be called here instead of constructing a Color from the ColorSpace type.